From 658881e640de392987fe7796f2fc254240bbdd62 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Wed, 23 Dec 2020 13:04:24 +0100
Subject: Fix some build problems on 32bit systems.

Eventually I hit a compiler error. Not sure what to do next.

diff --git a/Editor/ExportMedia_MediaKit.cpp b/Editor/ExportMedia_MediaKit.cpp
index 48b2e7c..4b4a5a8 100644
--- a/Editor/ExportMedia_MediaKit.cpp
+++ b/Editor/ExportMedia_MediaKit.cpp
@@ -178,7 +178,7 @@ void Export_MediaKit :: BuildVideoCodecOptions()
 		if (cookie != selected_cookie)
 			continue;
 
-		int cookie2 = 0;
+		int32 cookie2 = 0;
 		memset(&format, 0, sizeof(media_format));
 		format.type = B_MEDIA_RAW_VIDEO;
 		format.u.raw_video = media_raw_video_format::wildcard;
@@ -230,7 +230,7 @@ void Export_MediaKit :: BuildAudioCodecOptions()
 		if (cookie != selected_cookie)
 			continue;
 
-		int cookie2 = 0;
+		int32 cookie2 = 0;
 		memset(&format, 0, sizeof(media_format));
 		format.type = B_MEDIA_RAW_AUDIO;
 		while (get_next_encoder(&cookie2, &mfi, &format, &outfmt, &mci) == B_OK)
diff --git a/Editor/ExportMedia_ffmpeg.cpp b/Editor/ExportMedia_ffmpeg.cpp
index 3dd2bd6..ab07adf 100644
--- a/Editor/ExportMedia_ffmpeg.cpp
+++ b/Editor/ExportMedia_ffmpeg.cpp
@@ -960,7 +960,7 @@ void Export_ffmpeg :: StopEncode(const bool complete)
 
 /**************************************************************/
 /* WorkThread */
-int Export_ffmpeg :: WorkThread(void *arg)
+status_t Export_ffmpeg :: WorkThread(void *arg)
 {
 	assert(arg != nullptr);
 	Export_ffmpeg *instance = (Export_ffmpeg *) arg;
diff --git a/Editor/MonitorControls.cpp b/Editor/MonitorControls.cpp
index fe26cc0..e3bdcba 100644
--- a/Editor/MonitorControls.cpp
+++ b/Editor/MonitorControls.cpp
@@ -255,7 +255,7 @@ void MonitorControls :: DrawAfterChildren(BRect frame)
 		char buffer[20];
 		gProject->CreateTimeString(fCurrentFrame, buffer);
 		char *strarray[1] = {buffer};
-		int len[1] = {20};
+		int32 len[1] = {20};
 		float width;
 		GetStringWidths(strarray, len, 1, &width);
 
diff --git a/Editor/SourceListView.cpp b/Editor/SourceListView.cpp
index f6ab2ad..0123853 100644
--- a/Editor/SourceListView.cpp
+++ b/Editor/SourceListView.cpp
@@ -456,7 +456,7 @@ bool SourceListView :: AddItem(BListItem* item)
 {
 	if (fInstructionItemVisible)
 	{
-		InstructionListItem *item = (InstructionListItem *)RemoveItem(0);
+		InstructionListItem *item = (InstructionListItem *)RemoveItem((int32)0);
 		delete item;
 		fInstructionItemVisible = false;
 	}
@@ -474,6 +474,6 @@ void SourceListView :: RemoveAllMediaSources()
 	{
 		SourceListItem *item = (SourceListItem *)ItemAt(0);
 		gProject->RemoveMediaSource(item->GetMediaSource());
-		RemoveItem(0);
+		RemoveItem((int32)0);
 	}
 }
diff --git a/Editor/TimelinePosition.cpp b/Editor/TimelinePosition.cpp
index 875b785..feb184a 100644
--- a/Editor/TimelinePosition.cpp
+++ b/Editor/TimelinePosition.cpp
@@ -420,7 +420,7 @@ void TimelinePosition :: Draw(BRect frame)
 		char buffer[20];
 		gProject->CreateTimeString(fCurrentPosition, buffer, true);
 		char *strarray[1] = {buffer};
-		int len[1] = {20};
+		int32 len[1] = {20};
 		float width;
 		GetStringWidths(strarray, len, 1, &width);
 
diff --git a/Effects/Effect_Mask.cpp b/Effects/Effect_Mask.cpp
index 1a6c94e..96dbf20 100644
--- a/Effects/Effect_Mask.cpp
+++ b/Effects/Effect_Mask.cpp
@@ -584,7 +584,7 @@ void Effect_Mask :: MessageReceived(BMessage *msg)
 
 		case kMsgKeyframeSlider:
 		{
-			int selection;
+			int32 selection;
 			if (msg->FindInt32("selection", &selection) == B_OK)
 			{
 				assert((selection >= 0) && (selection < fKeyframeList->CountItems()));
diff --git a/Gui/FontPanel.cpp b/Gui/FontPanel.cpp
index f5a082d..b76a922 100644
--- a/Gui/FontPanel.cpp
+++ b/Gui/FontPanel.cpp
@@ -439,7 +439,7 @@ void FontView :: CreateStyleList(int32 family_idx)
 
 	while (fStyleList->CountItems() > 0)
 	{
-		BStringItem *item = (BStringItem*)fStyleList->RemoveItem(0);
+		BStringItem *item = (BStringItem*)fStyleList->RemoveItem((int32)0);
 		delete item;
 	}
 
diff --git a/Jamfile b/Jamfile
index 839b0b7..45eee66 100644
--- a/Jamfile
+++ b/Jamfile
@@ -11,6 +11,7 @@ else
 }
 C++FLAGS += -I. -IYarra -I/boot/system/develop/headers/private/interface ;
 C++FLAGS += -I/boot/system/develop/headers/freetype2 ;
+C++FLAGS += -I/boot/system/develop/headers/x86/freetype2 -mavx ;
 
 # Source files needed to build Medo
 # <pe-src>
-- 
2.28.0

